internal/gitprovider/azure/azure.go

```diff
@@ -0,0 +1,234 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"slices"
+	"strings"
+
+	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/gitprovider"
+	"k8s.io/utils/ptr"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7"
+	adogit "github.com/microsoft/azure-devops-go-api/azuredevops/v7/git"
+)
+
+const ProviderName = "azure"
+
+// Azure DevOps URLs can be of two different forms:
+//
+//   - https://dev.azure.com/org/project/_git/repo
+//   - https://org.visualstudio.com/project/_git/repo
+//
+// We support both forms.
+var providerSuffixes = []string{"dev.azure.com", "visualstudio.com"}
+
+var registration = gitprovider.Registration{
+	Predicate: func(repoURL string) bool {
+		u, err := url.Parse(repoURL)
+		if err != nil {
+			return false
+		}
+		return slices.ContainsFunc(providerSuffixes, func(suffix string) bool {
+			return strings.HasSuffix(u.Host, suffix)
```

Comment:
Is this incorrectly tolerant of `foobar.dev.azure.com`, which I presume to be invalid?

It may not matter much, so I consider this non-blocking.

Reply:
done